types/node/v16/module.d.ts

```diff
@@ -195,8 +195,13 @@ declare module "module" {
         static Module: typeof Module;
         constructor(id: string, parent?: Module);
     }
+    type ImportMetaDOMCompat = typeof globalThis extends { onmessage: any } ? {
+    resolve(specifier: string): string;
```

Comment:
I feel like this is going to be a hazard; if anyone includes DOM, they're going to be misled into thinking they don't need to await. But, if they're including DOM, maybe their "main" environment is DOM and they don't want to await?

I guess I don't know how to do any better.

Reply:
I think people with lib.dom.d.ts and @types/node are more likely to be running in the DOM than in Node.js. I.e., it’s easier to include @types/node unintentionally than it is to include the DOM lib unintentionally.